src/Core/Validation/ValidationException.php

```diff
@@ -48,6 +52,16 @@ public function __construct($result = null, $code = 0)
             // Pick first message
             foreach ($result->getMessages() as $message) {
                 $exceptionMessage = $message['message'];
+                if ($this->doShowAdditionalInfo()) {
+                    if ($message['fieldName']) {
+                        $exceptionMessage .= ' - fieldName: ' . $message['fieldName'];
+                    }
+                    $dataClass = $result->getDataClass();
+                    if (is_subclass_of($dataClass, DataObject::class, true)) {
+                        $exceptionMessage .= ', recordID: ' . $result->getRecordID();
+                        $exceptionMessage .= ', dataClass: ' . $dataClass;
+                    }
+                }
```

Comment:
Is there ever a reason not to include this additional info in the exception message? e.g. where `ValidationException` gets thrown for a CMS form submission and `$result` is a `ValidationResult`, the actual message isn't parsed by the UI - instead the validation result's individual messages are included in JSON in the response and are used appropriately.

Reply:
As mentioned in https://github.com/silverstripe/silverstripe-framework/pull/11562#discussion_r1945601883 I think we need to keep `doShowAdditionalInfo()`

Reply:
Will resolve this as it will be obvious based on the discussion in that other thread whether changes are needed here or not.